pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/service/api/Component.java

```diff
@@ -43,7 +43,21 @@ void deregisterFunction(final String tenant,
                             final String namespace,
                             final String componentName,
                             final String clientRole,
-                            AuthenticationDataHttps clientAuthenticationDataHttps);
+                            AuthenticationDataSource clientAuthenticationDataHttps);
+
+    @Deprecated
+    default void deregisterFunction(final String tenant,
```

Comment:
Set this method to deprecated state, in order to maintain backward compatibility